src/Cms/Collection.php

```diff
@@ -321,6 +322,32 @@ public function search(string $query = null, $params = [])
         return Search::collection($this, $query, $params);
     }
 
+    /**
+     * Sorts the elements by any number of fields and ensures that
+     * no write operations can happen by disabling all permissions
+     *
+     * @param string|callable $field Field name or value callback to sort by
+     * @param string $direction asc or desc
+     * @param int $method The sort flag, SORT_REGULAR, SORT_NUMERIC etc.
+     * @return \Kirby\Cms\Collection
+     */
+    public function sortBy()
+    {
+        $auth = kirby()->auth();
+        $userBefore = $auth->currentUserFromImpersonation();
+
+        try {
+            $auth->impersonate('nobody');
+            return parent::sortBy(...func_get_args());
+        } catch (Throwable $e) {
+            throw $e;
```

Comment:
Maybe we should provide a custom error message if it is a PermissionException? Instead of something like "insufficient permissions" we could let the user know that they are falsely using the name of an altering method instead of the field they likely wanted to use.